### PR TITLE
XMLHttpRequest/File API: fetch revoked blob URL

### PR DESCRIPTION
**Pull request by annevk**
Closes https://github.com/whatwg/xhr/issues/79.

### CODE REVIEW

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/204126101)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 048e168bef95512cd7c587602531c807c28cab87
Using browser at version BuildID 20170221110300; SourceStamp d84beb192e57e26846c82d3df3599381f4663792
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/4953/FileAPI/blob/Blob-XHR-revoke.html">/FileAPI/blob/Blob-XHR-revoke.html</a></summary>


|                    Subtest                     | Results |           Messages          |
|------------------------------------------------|---------|-----------------------------|
|                                                | OK      |                             |
| `Revoking blob URL used with XMLHttpRequest`   | PASS    |                             |
| `Revoking blob URL used with XMLHttpRequest 1` | FAIL    | `A network error occurred.` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/204126102)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 048e168bef95512cd7c587602531c807c28cab87
Using browser at version 58.0.3013.3 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/4953/FileAPI/blob/Blob-XHR-revoke.html">/FileAPI/blob/Blob-XHR-revoke.html</a></summary>


|                    Subtest                     | Results |                                Messages                                |
|------------------------------------------------|---------|------------------------------------------------------------------------|
|                                                | OK      |                                                                        |
| `Revoking blob URL used with XMLHttpRequest`   | FAIL    | `assert_equals: expected "test" but got ""`                            |
| `Revoking blob URL used with XMLHttpRequest 1` | FAIL    | `assert_unreached: Got unexpected load event Reached unreachable code` |

</details>
</details>

File: FileAPI/blob/Blob-XHR-revoke.html [hunk 1]
```diff
@@ -17,12 +17,22 @@
...
+async_test(t => {
+  const blob = new Blob(["test"]),
+        blobURL = URL.createObjectURL(blob),
+        client = new XMLHttpRequest
+  URL.revokeObjectURL(blobURL)
+  client.open("GET", blobURL)
+  client.onload = t.step_func(() => assert_unreached("Got unexpected load event"))
+  client.onerror = t.step_func_done()
+  client.send()
+})
```
Comment (jdm) [inline FileAPI/blob/Blob-XHR-revoke.html]: This (and the previous async_test) now need explicit test descriptions.

**Review by jdm — CHANGES_REQUESTED** [review]
Apart from the one comment this looks fine.

File: FileAPI/blob/Blob-XHR-revoke.html (hunk 1 above)
Comment (annevk) [inline FileAPI/blob/Blob-XHR-revoke.html]: Isn't the default 1, 2 okay? Anyway, I guess I can add something.

**Pull request merged**